bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/ICalendarHandler.java

```diff
@@ -162,11 +164,29 @@ public void initialize() {
     public void onCalendarUpdated() {
         if (reloadCalendar()) {
             updateStates();
+            for (Thing childThing : getThing().getThings()) {
+                ThingHandler handler = childThing.getHandler();
+                if (handler instanceof CalendarUpdateListener) {
+                    try {
+                        ((CalendarUpdateListener) handler).onCalendarUpdated();
+                    } catch (Exception e) {
```

Comment:
What type of exception do you expect here? Can you specify the concrete type?

Reply:
I can't. This catch clause was added for resiliency against failures in other classes inside the callback. Other callbacks should not be affected if a callback fails.